src/hooks/provider/index.ts

```diff
@@ -0,0 +1,47 @@
+import { useCallback, useMemo } from "react";
+import { useEthereumTransactionState, TransactionStateStatus } from "../helpers";
+import { ethers, providers } from "ethers";
+
+import { getLogger } from "../../utils/logger";
+const { trace, error } = getLogger("hook:etherjs:provider");
+
+interface Ethereum extends providers.Web3Provider {
+  enable: () => void;
+}
+
+declare global {
```

Comment:
there is no typescript type ? If no move this to a specific type file

Reply:
moved to common declaration.d.ts